### PR TITLE
ROADMAP: Update roadmap

### PR DESCRIPTION
**Pull request by shenli**
The roadmap is out of date. So we need to update it.
@coocood @lilin90 

Related with https://github.com/pingcap/docs-cn/pull/762

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -10,21 +10,30 @@ This document defines the roadmap for TiDB development.
...
-- [ ] Support Window Function
+- [ ] View
+- [ ] Window Function
 - [ ] Common Table Expression
 - [ ] Table Partition
-- [ ] Hash time index to resolve the issue with hot regions
-- [ ] Reverse Index
 - [ ] Cluster Index
 - [ ] Improve DDL
+  - [ ] Speed up Add Index operation
```
Comment (lilin90) [inline ROADMAP.md]: This one is marked as done in docs-cn. Please keep it consistent. https://github.com/pingcap/docs-cn/pull/762/files

**Comment by shenli** [comment]
@lilin90 PTAL

File: ROADMAP.md [hunk 2]
```diff
@@ -10,21 +10,30 @@ This document defines the roadmap for TiDB development.
...
 - [ ] Optimizer
-  - [ ] Refactor Ranger
-  - [ ] Optimize the statistics info 
+  - [x] Refactor Ranger
   - [ ] Optimize the cost model
+  - [ ] Join Reorder
+- [ ] Statistics
+  - [x] Update statistics dynamically according to the query feedback
+  - [x] Analyze table automatically
+  - [ ] Improve the accuracy of Row Cout estimation
```
Comment (lilin90) [inline ROADMAP.md]: Row Cout -> Row Count

File: ROADMAP.md [hunk 3]
```diff
@@ -10,21 +10,30 @@ This document defines the roadmap for TiDB development.
...
 - [ ] Executor
+  - [ ] Pushdown the Projection operator to the Coprocessor
```
Comment (lilin90) [inline ROADMAP.md]: Pushdown -> Push down

Note: Use the two words version as a verb. "pushdown" is usually used as a noun.

**Comment by shenli** [comment]
@lilin90 PTAL

**Pull request merged**